Lib/ast.py

```diff
@@ -368,6 +368,69 @@ def walk(node):
         yield node
 
 
+def compare(
+    a,
+    b,
+    /,
+    *,
+    compare_types=True,
+    compare_fields=True,
+    compare_attributes=False,
+):
+    """
+    Recursively compare given 2 AST nodes. If *compare_types* is False, the
+    field values don't be checked whater they belong to same type or not. If
+    *compare_fields* is True, members of `_fields` attribute on both node's type
+    will be checked. If *compare_attributes* is True, members of `_attributes`
+    attribute on both node's will be compared.
+    """
+
+    def _compare(a, b):
+        if compare_types and type(a) is not type(b):
+            return False
+        elif isinstance(a, AST):
+            return compare(
+                a,
+                b,
+                compare_attributes=compare_attributes,
+                compare_fields=compare_fields,
+                compare_types=compare_types,
+            )
+        elif isinstance(a, list):
+            if len(a) != len(b):
+                return False
+            for a_item, b_item in zip(a, b):
+                if not _compare(a_item, b_item):
+                    return False
+            else:
```

Comment:
We can remove this else and move return statement parallel to for...
At other places also

Reply:
I disagree, the whole purpose of `else` is this.